<!-- Launch file for the rrt-detector and the assigner -->


<launch>
<arg name="eta" default="25.0"/>
<arg name="Geta" default="50.0"/>
<arg name="info_radius" default="150.0"/>
<arg name="rate" default="1.0"/>

<param name="namespace_init_count" value="1"/>


  <node pkg="rrt_exploration" type="global_rrt_detector" name="global_detector" output="screen">
  <param name="eta" value="$(arg Geta)"/>

  <remap from="map" to="lio_sam/mapping/map_global/map"/>
  </node>
  
  <node pkg="rrt_exploration" type="local_rrt_detector" name="local_detector" output="screen">
  <param name="eta" value="$(arg eta)"/>
  <param name="map_topic" value="lio_sam/mapping/map_global/map"/>
  <param name="robot_frame" value="base_link"/>
  </node>
 
  
  <node pkg="rrt_exploration" type="filter.py" name="filter" output="screen">
  <param name="info_radius" value="$(arg info_radius)"/> 
  <param name="robot_frame" value="base_link"/>
  <param name="costmap_clearing_threshold" value="70"/> 
  <param name="cluster_mean_shift_bandwith" value="-1" />
  <param name="rate" value="$(arg rate)"/>

  <remap from="map" to="lio_sam/mapping/map_global/map"/>
  <remap from="costmap" to="move_base_flex/global_costmap/costmap"/>
  <remap from="detected_points" to="detected_points"/>
  </node>
  
  <node pkg="rrt_exploration" type="assigner.py" name="assigner" output="screen">
  <param name="global_frame" value="map"/>
  <param name="info_radius" value="$(arg info_radius)"/> 
  <param name="info_multiplier" value="3.0"/> 
  <param name="hysteresis_radius" value="40.0"/> 
  <param name="hysteresis_gain" value="2.0"/> 
  <param name="delay_after_assignement" value="0.5"/>
  <param name="rate" value="$(arg rate)"/>

  <remap from="map" to="lio_sam/mapping/map_global/map"/>
  <remap from="filtered_points" to="filtered_points"/>
  </node>
  
</launch>
